client/state/data-layer/analytics/adTracking.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	trackCustomAdWordsRemarketingEvent,
+	trackCustomFacebookConversionEvent,
+} from 'lib/analytics/ad-tracking';
+import { SIGNUP_STEPS_SURVEY_SET } from 'state/action-types';
+
+function recordSelectedVertical( _, action, next ) {
+	next( action );
```

Comment:
this would work!

@gwwar you mean adding some new analytics meta information to the action in `client/state/signup/steps/survey/actions.js` for fb ads and google adwords then some new event services to `client/state/analytics/middleware.js`?

As long as the components themselves no longer need to know about firing analytics events, this sounds great to me and avoids the data layer problem mentioned by @dmsnell.